### DISCUSSION TITLE
My problem is with the new package (50.2.4.2) and exporting the application

### DISCUSSION THREAD

**Discussion by avcr2019** [airsdk/Adobe-Runtime-Support#3016] [General]
Welcome my friends
I have a strange problem and I don't know its cause and I don't know why it happens in the first place.
I am using Harman package version 50.2.3.6 and it is working well for me so far without any problems.
But I have been absent for a while and by chance I found that there are newer and better versions, so I downloaded the latest version 50.2.4.2 and tried it on the computer and it works successfully and the file is exported to (APK) successfully, but the problem occurs from here after exporting the file, which is that when I I tried the file on my phone but it did not install and it did. There was a problem while installing the package.
![1_Chj323V_MeL5rT9jrYC3WQ_o](https://github.com/airsdk/Adobe-Runtime-Support/assets/51138931/64be7772-4cf4-4d1c-9268-61d7e2cdb251)

 Knowing that the file was exported successfully and without any errors. Here is the file for package 50.2.4.2
![Screenshot 2024-01-20 at 3 16 27 AM](https://github.com/airsdk/Adobe-Runtime-Support/assets/51138931/0eae7fc9-235f-4a4e-97cd-ee4ee51e04bb)
As you can see, I did not choose for it to do (CreateAndroidAppBundle), I just let it export to (APK).
On the other hand, when I export the same file with the same settings on package 50.2.3.6, it exports the file to (APK) and it works successfully and is installed without any problems. Here I don't know why, knowing that I am using the latest version of the program + that I have updated all the packages inside Android Studio.
Will I still be stuck on this version 50.2.3.6?? I want to be with the new and fresh from the pack.
Note: I don't know if it has a relationship or not. I put this code in the application's ghola file:
`  <gradleVersion>7.5</gradleVersion>
        <androidGradlePluginVersion>7.4.0</androidGradlePluginVersion>`
I don't know if it has anything to do with it or not, especially since I didn't change it and when I updated it and made it like this
`  <gradleVersion>8.0</gradleVersion>
        <androidGradlePluginVersion>8.0.2</androidGradlePluginVersion>`
  It didn't work either so I'm confused?
When do I update the gradleVersion?
I mean, does this have much effect when exporting the application?
I mean here, is there a problem if I make it as it was before or do I update it?

**Comment by FliplineStudios** [comment]
There was actually a patch for 50.2.4.2 that dealt with signing issues where apps couldn't be installed on Android devices, it may be that this patch isn't included in the main SDK yet.  You can find the patch for adt in the comment here:

https://github.com/airsdk/Adobe-Runtime-Support/issues/2978#issuecomment-1866848279

I'd give this a try first to see if it makes a difference!

**Reply by avcr2019** [reply]
Well thank you, my dear friend
I have tried the file and the application has already been exported and is working successfully on the phone now. But I have one comment: When I export the application to (AAB), do I use this same new file? Or do I use the old file that was present, and here I mean this file (adt.jar).
When exporting, I saw this warning. Is it effective or is it something normal because it has never happened to me before:
![Screenshot 2024-01-20 at 12 51 57 PM](https://github.com/airsdk/Adobe-Runtime-Support/assets/51138931/30186896-f5ee-4e23-8fd0-5993afaf8b69)

- Do you have any idea when I will update this(gradleVersion)?
![Screenshot 2024-01-20 at 2 31 43 AM](https://github.com/airsdk/Adobe-Runtime-Support/assets/51138931/b668eea3-769d-49c1-940f-4f4db52c7226)

Should I leave it as it is or should I update it?

**Reply by avcr2019** [reply]
amendment :
I exported to (AAB) using the new file (adt.jar), but unfortunately the package was exported in a small size of 116 MB!!! This is considered a problem because if the file itself is exported to (APK) only, its size is 126 MB. This indicates that there is a problem with exporting to (AAB) using this new package 50.2.4.2. While when I exported with the old package 50.2.3.6, the file was exported to (AAB) and had a size of 149 MB, and this is the correct number. Will we still be using old packages?? When will these problems be resolved in the new packages?

**Reply by FliplineStudios** [reply]
If you're using Animate, I think there are some other issues specific to Animate with packaging AAB, there was some discussion about some things you may need to set here: #2992 

That first warning about "Multiple Android build-tools folders exist" is normal, it just means you have more than one Android SDK version available to use for packaging, and it's choosing the latest one (we normally see the same warning when packaging from the command line).

I'm not really sure about updating the gradleVersion, I don't believe we've ever defined it in manifestAdditions ourselves.  Personally I don't think I'd worry about changing it unless there was a problem with packaging where you needed to use a newer version (there was also mention of how to switch to Gradle 8.4 if needed here: #2965)

**Reply by avcr2019** [reply]
Thank you for your interest, my friend, and thank you for the beautiful advice you mentioned in your message
Well, I will read here (#2992) and I hope I find the correct answer
